lib/modules/datasource/go/base.ts

```diff
@@ -173,18 +180,26 @@ export class BaseGoDatasource {
     // get server base url from import url
     const parsedUrl = URL.parse(goImportURL);
 
-    // split the go module from the URL: host/go/module -> go/module
-    // TODO: `parsedUrl.pathname` can be undefined
-    const packageName = trimTrailingSlash(`${parsedUrl.pathname}`)
-      .replace(regEx(/\.git$/), '')
-      .split('/')
-      .slice(-2)
-      .join('/');
-
-    return {
-      datasource: GithubTagsDatasource.id,
-      registryUrl: `${parsedUrl.protocol}//${parsedUrl.host}`,
-      packageName,
-    };
+    const datasource = this.detectDatasource(
+      goImportURL.replace(regEx(/\.git$/), ''),
+      goModule
+    );
+    // fall back to old behaviour if detection did not work
+    if (datasource === null) {
```

Comment:
```suggestion
    if (datasource !== null) {
      return datasource;
    }
```

Return early to reduce white space